client_cli/src/main.rs

```diff
@@ -55,33 +55,29 @@ impl MetadataArgs {
     }
 }
 
-/// Re-usable clap `--value <Value>` (`-v`) argument.
+/// Re-usable clap `--value <MetadataValue>` (`-v`) argument.
 /// Should be combined with `#[command(flatten)]` attr.
 #[derive(clap::Args, Debug, Clone, PartialEq, Eq)]
-pub struct ValueArg {
-    /// Wrapper around Value to accept possible values and fallback to json.
+pub struct MetadataValueArg {
+    /// Wrapper around MetadataValue to accept possible values and fallback to json.
+    ///
     /// The following types are supported:
     /// Numbers: with a suffix, e.g. 42_u32 or 1000_u128
     /// Booleans: false/true
-    /// IPv4/IPv6: e.g. 127.0.0.1, ::1
-    /// Iroha Public Key Multihash: e.g. ed01207233BFC89DCBD68C19FDE6CE6158225298EC1131B6A130D1AEB454C1AB5183C0
     /// JSON: e.g. {"Vec":[{"String":"a"},{"String":"b"}]}
     #[arg(short, long)]
-    value: Value,
+    value: MetadataValue,
 }
 
-impl FromStr for ValueArg {
+impl FromStr for MetadataValueArg {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         s.parse::<bool>()
-            .map(Value::Bool)
-            .or_else(|_| s.parse::<Ipv4Addr>().map(Value::Ipv4Addr))
-            .or_else(|_| s.parse::<Ipv6Addr>().map(Value::Ipv6Addr))
-            .or_else(|_| s.parse::<NumericValue>().map(Value::Numeric))
-            .or_else(|_| s.parse::<PublicKey>().map(Value::PublicKey))
-            .or_else(|_| serde_json::from_str::<Value>(s).map_err(Into::into))
-            .map(|value| ValueArg { value })
+            .map(MetadataValue::Bool)
+            .or_else(|_| s.parse::<NumericValue>().map(MetadataValue::Numeric))
+            .or_else(|_| serde_json::from_str::<MetadataValue>(s).map_err(Into::into))
```

Comment:
idk, that's how it was before my work. @0x009922 maybe you can tell us?

Reply:
I guess it is because if it is deserialized directly using `Value` (or `MetadataValue` now), it requires specifying a full enum repr in CLI, e.g.:

```shell
iroha_client_cli ... --value='{"Bool":false}'
```

Doesn't look CLI friendly, does it? So, in https://github.com/hyperledger/iroha/pull/4163 @Asem-Abdelhady added this chain parsing which allows a simplified form:

```shell
iroha_client_cli ... --value=false
```